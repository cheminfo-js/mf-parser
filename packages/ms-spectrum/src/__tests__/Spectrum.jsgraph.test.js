'use strict';

const Spectrum = require('../Spectrum');

describe('test Spectrum JSGraph', () => {
  it('getAnnotation', () => {
    let peaks = [
      {
        x: 1,
        y: 2,
        close: false,
      },
      { x: 2, y: 3, close: true },
    ];
    let annotations = Spectrum.JsGraph.getPeaksAnnotation(peaks);

    expect(annotations).toHaveLength(3);
    expect(annotations).toMatchSnapshot();
  });
});
