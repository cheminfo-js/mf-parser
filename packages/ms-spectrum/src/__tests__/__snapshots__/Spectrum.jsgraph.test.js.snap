// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Spectrum JSGraph getAnnotation 1`] = `
Array [
  Object {
    "_highlight": undefined,
    "labels": Array [
      Object {
        "color": "red",
        "position": Object {
          "dx": "2px",
          "dy": "-17px",
          "x": 1,
          "y": 2,
        },
        "text": "1.00000",
      },
    ],
    "position": Array [
      Object {
        "dy": "-5px",
        "x": 1,
        "y": 2,
      },
      Object {
        "dy": "-25px",
        "x": 1,
        "y": 2,
      },
    ],
    "type": "line",
  },
  Object {
    "_highlight": undefined,
    "info": Object {
      "close": true,
      "x": 2,
      "y": 3,
    },
    "position": Array [
      Object {
        "dy": "-5px",
        "x": 2,
        "y": 3,
      },
      Object {
        "dy": "-10px",
        "x": 2,
        "y": 3,
      },
    ],
    "type": "line",
  },
]
`;
